cluster-tests/tests/cleanup/test_cluster_cleanup.py

```diff
@@ -0,0 +1,115 @@
+from time import sleep
+
+import pytest
+from pytest_dependency import depends
+
+from config import CLUSTER_INFO, CLUSTER_IPS
+from utils import make_request, make_ws_request, wait_on_job
+from exceptions import JobTimeOut
+
+IGNORE = ['ctdb_shared_vol']  # TODO: dont hardcode this
+
+
+@pytest.mark.dependency(name='STOP_GVOLS')
+def test_stop_all_gvols():
+    ans = make_request('get', '/gluster/volume')
+    assert ans.status_code == 200, ans.text
+
+    # we will try to stop each gluster volume `retries` times waiting at least
+    # 1 second between each attempt
+    retries = 20
+    sleeptime = 1
+    for i in filter(lambda x: x['status'] == 'Started' and x['name'] not in IGNORE, ans.json()):
+        gvol = i['name']
+        for retry in range(retries):
+            stop = make_request('post', '/gluster/volume/stop', data={'name': gvol, 'force': True})
+            if stop.status_code == 422:
+                if 'Another transaction is in progress' in stop.text:
+                    # another command is running in the cluster so this isn't fatal but expected
+                    # so we'll sleep for `sleeptime` seconds to backoff and let cluster settle
+                    sleep(sleeptime)
+                else:
+                    assert False, f'Failed to stop {gvol!r}: {stop.text}'
+            elif stop.status_code == 200:
+                break
+            elif retry == retries:
+                assert False, f'Retried {retries} times to stop {gvol!r} but failed.'
+
+
+@pytest.mark.parametrize('ip', CLUSTER_IPS)
+@pytest.mark.dependency(name='VERIFY_FUSE_UMOUNTED')
+def test_verify_all_gvols_are_fuse_umounted(ip, request):
+    depends(request, ['STOP_GVOLS'])
+
+    ans = make_request('get', '/gluster/volume/list')
+    assert ans.status_code == 200, ans.text
+
+    # we will try to check if each FUSE mountpoint is umounted `retries` times waiting at
+    # least `sleeptime` second between each attempt
+    retries = 10
```

Comment:
Should we maybe use the FUSE_OP_TIMEOUT from config? That way it can be tweaked if needed via environmental variable?